tests/MqttRequestResponse.cpp

```diff
@@ -0,0 +1,1099 @@
+{
+    {
+        std::unique_lock<std::mutex> lock(state->lock);
+        state->subscriptionStatusEvents.push_back(event);
+    }
+    state->signal.notify_one();
+}
+
+static void s_waitForSubscriptionStatusEvent(
+    TestState *state,
+    Aws::Iot::RequestResponse::SubscriptionStatusEventType type,
+    int errorCode)
+{
+    {
+        std::unique_lock<std::mutex> lock(state->lock);
+        state->signal.wait(
+            lock,
+            [=]()
+            {
+                return std::any_of(
+                    state->subscriptionStatusEvents.cbegin(),
+                    state->subscriptionStatusEvents.cend(),
+                    [=](const Aws::Iot::RequestResponse::SubscriptionStatusEvent &event)
+                    { return event.GetType() == type && event.GetErrorCode() == errorCode; });
+            });
+    }
+}
+
+static void s_onIncomingPublishEvent(Aws::Iot::RequestResponse::IncomingPublishEvent &&event, TestState *state)
+{
+    {
+        std::unique_lock<std::mutex> lock(state->lock);
+
+        auto payloadCursor = event.GetPayload();
+        Aws::Crt::String payloadAsString((const char *)payloadCursor.ptr, payloadCursor.len);
+
+        state->incomingPublishEvents.push_back(payloadAsString);
+    }
+    state->signal.notify_one();
+}
+
+static void s_waitForIncomingPublishWithPredicate(
+    TestState *state,
+    const std::function<bool(const Aws::Crt::String &)> &predicate)
+{
+    {
+        std::unique_lock<std::mutex> lock(state->lock);
+        state->signal.wait(
+            lock,
+            [=]()
+            {
+                return std::any_of(
+                    state->incomingPublishEvents.cbegin(),
+                    state->incomingPublishEvents.cend(),
+                    [=](const Aws::Crt::String &payload) { return predicate(payload); });
+            });
+    }
+}
+
+struct TestContext
+{
+    std::shared_ptr<Aws::Iot::RequestResponse::IMqttRequestResponseClient> client;
+    std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> protocolClient5;
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> protocolClient311;
+};
+
+static void s_startProtocolClient(TestContext &context)
+{
+    if (context.protocolClient5)
+    {
+        context.protocolClient5->Start();
+    }
+    else
+    {
+        auto uuid = Aws::Crt::UUID().ToString();
+        context.protocolClient311->Connect(uuid.c_str(), true, 30, 15000, 5000);
+    }
+}
+
+static TestContext s_CreateClient(
+    Aws::Crt::Allocator *allocator,
+    ProtocolType protocol,
+    struct TestState *state,
+    Aws::Iot::RequestResponse::RequestResponseClientOptions *options = NULL)
+{
+    TestContext context;
+
+    Aws::Iot::RequestResponse::RequestResponseClientOptions finalOptions;
+    struct aws_string *host = NULL;
+    struct aws_string *certificatePath = NULL;
+    struct aws_string *privateKeyPath = NULL;
+
+    if (aws_get_environment_value(allocator, s_rrEnvVariableHost, &host))
```

Comment:
`aws_get_environment_value` returns success on missing env var (kinda counterintuitive, I'd say).
As a side note: in the iot tests, I implemented a wrapper function returning `Optional<String>`: https://github.com/awslabs/aws-crt-cpp/blob/main/tests/IotServiceTest.cpp#L50